test/controllers/api/notes_controller_test.rb

```diff
@@ -200,14 +202,44 @@ def test_create_fail
       assert_response :bad_request
     end
 
+    def test_create_success
+      user = create(:user)
+      auth_header = bearer_authorization_header user
+      assert_difference "Note.count", 1 do
+        assert_difference "NoteComment.count", 1 do
+          assert_difference "NoteSubscription.count", 1 do
+            post api_notes_path(:lat => -1.0, :lon => -1.0, :text => "This is a comment", :format => "json"), :headers => auth_header
+          end
+        end
+      end
+      assert_response :success
+      js = ActiveSupport::JSON.decode(@response.body)
+      assert_not_nil js
+      assert_equal "Feature", js["type"]
+      assert_equal "Point", js["geometry"]["type"]
+      assert_equal [-1.0, -1.0], js["geometry"]["coordinates"]
+      assert_equal "open", js["properties"]["status"]
+      assert_equal 1, js["properties"]["comments"].count
+      assert_equal "opened", js["properties"]["comments"].last["action"]
+      assert_equal "This is a comment", js["properties"]["comments"].last["text"]
+      assert_equal user.display_name, js["properties"]["comments"].last["user"]
+
+      note = Note.last
+      subscription = NoteSubscription.last
```

Comment:
This (and all the other similar versions in other tests) is relying on nothing else having created notes or users in a way that overlaps this test and I'm not sure how safe that is when tests are running in parallel?

Maybe it would be better to get the node ID from the JSON response and then do `assert Note.exists?(id)` and a similar exists assertion on the subscription record?

Reply:
> I'm not sure how safe that is when tests are running in parallel?

Parallel tests are run in separate databases (e.g. `openstreetmap_test-0` etc)

Reply:
> This ... is relying on nothing else having created notes

There is a check `assert_difference "Note.count", 1`

> (and all the other similar versions in other tests)

Other tests in this file or other tests elsewhere? You can find similar uses of `.last` from years ago.